src/JustSaying/Fluent/QueueAddressSubscriptionBuilder`1.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using Amazon;
+using Amazon.SQS;
+using JustSaying.AwsTools;
+using JustSaying.AwsTools.QueueCreation;
+using JustSaying.Messaging.Middleware;
+using JustSaying.Models;
+using Microsoft.Extensions.Logging;
+
+namespace JustSaying.Fluent
+{
+    public sealed class QueueAddressSubscriptionBuilder<T> : ISubscriptionBuilder<T>
+        where T : Message
+    {
+        private readonly QueueAddress _queueAddress;
+
+        internal QueueAddressSubscriptionBuilder(QueueAddress queueAddress)
+        {
+            _queueAddress = queueAddress;
+        }
+
+        private Action<QueueAddressConfiguration> ConfigureReads { get; set; }
+
+        public QueueAddressSubscriptionBuilder<T> WithReadConfiguration(Action<QueueAddressConfiguration> configure)
+        {
+            ConfigureReads = configure ?? throw new ArgumentNullException(nameof(configure));
+            return this;
+        }
+
+        /// <inheritdoc />
+        void ISubscriptionBuilder<T>.Configure(
+            JustSayingBus bus,
+            IHandlerResolver handlerResolver,
+            IServiceResolver serviceResolver,
+            IVerifyAmazonQueues creator,
+            IAwsClientFactoryProxy awsClientFactoryProxy,
+            ILoggerFactory loggerFactory)
+        {
+            var logger = loggerFactory.CreateLogger<QueueSubscriptionBuilder<T>>();
+
+            var attachedQueueConfig = new QueueAddressConfiguration();
+
+            ConfigureReads?.Invoke(attachedQueueConfig);
+
+            IAmazonSQS sqsClient = awsClientFactoryProxy
+                .GetAwsClientFactory()
+                .GetSqsClient(RegionEndpoint.GetBySystemName(_queueAddress.RegionName));
+
+            var queue = new QueueAddressQueue(_queueAddress, sqsClient);
+
+            attachedQueueConfig.SubscriptionGroupName ??= queue.QueueName;
+            attachedQueueConfig.Validate();
+
+            bus.AddQueue(attachedQueueConfig.SubscriptionGroupName, queue);
+
+            logger.LogInformation(
+                "Created SQS queue subscription for '{MessageType}' on '{QueueName}'",
+                typeof(T), queue.QueueName);
+
+            var resolutionContext = new HandlerResolutionContext(queue.QueueName);
+            var proposedHandler = handlerResolver.ResolveHandler<T>(resolutionContext);
+            if (proposedHandler == null)
+            {
+                throw new HandlerNotRegisteredWithContainerException(
+                    $"There is no handler for '{typeof(T)}' messages.");
```

Comment:
Same here - I forget if this makes it format the assembly qualified name or not.

Reply:
They will get ToStringed, so for our purposes it's the same as `Fullname` (i.e. `Namespace.TypeName`), but I think we aren't being constant across the codebase, there are places we using `ToString`, `Fullname` and `Name` for seemingly the same thing.

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.type.fullname?view=net-5.0#code-try-3
^ differences between how they behave